controllers/vrg_kubeobjects.go

```diff
@@ -558,3 +663,283 @@ func kubeObjectsRequestsWatch(b *builder.Builder, kubeObjects kubeobjects.Reques
+			if err != nil {
+				return resources, err
+			}
+
+			resources = append(resources, *captureInstance)
+		}
+	}
+
+	return resources, nil
+}
+
+func getResourceAndConvertToCaptureGroup(
+	recipe Recipe.Recipe, resourceType, name string) (*ramen.KubeObjectsCaptureSpec, error,
+) {
+	// check hooks OR groups
+	if resourceType == "group" {
+		for _, group := range recipe.Spec.Groups {
+			if group.Name == name {
+				return convertRecipeGroupToCaptureSpec(*group)
+			}
+		}
+
+		return nil, k8serrors.NewNotFound(schema.GroupResource{Resource: "Recipe.Spec.Group.Name"}, name)
+	}
+
+	if resourceType == "hook" {
+		hook, op, err := getHookAndOpFromRecipe(&recipe, name)
+		if err != nil {
+			return nil, k8serrors.NewNotFound(schema.GroupResource{Resource: "Recipe.Spec"}, resourceType)
+		}
+
+		return convertRecipeHookToCaptureSpec(*hook, *op)
+	}
+
+	return nil, k8serrors.NewNotFound(schema.GroupResource{Resource: "Recipe.Spec"}, resourceType)
+}
+
+// resource: could be Group or Hook
+func getResourceAndConvertToRecoverGroup(
+	recipe Recipe.Recipe, resourceType, name string) (*ramen.KubeObjectsRecoverSpec, error,
+) {
+	if resourceType == "group" {
+		for _, group := range recipe.Spec.Groups {
+			if group.Name == name {
+				return convertRecipeGroupToRecoverSpec(*group)
+			}
+		}
+
+		return nil, k8serrors.NewNotFound(schema.GroupResource{Resource: "Recipe.Spec.Group.Name"}, name)
+	}
+
+	if resourceType == "hook" {
+		hook, op, err := getHookAndOpFromRecipe(&recipe, name)
+		if err != nil {
+			return nil, k8serrors.NewNotFound(schema.GroupResource{Resource: "Recipe.Spec"}, resourceType)
+		}
+
+		return convertRecipeHookToRecoverSpec(*hook, *op)
+	}
+
+	return nil, k8serrors.NewNotFound(schema.GroupResource{Resource: "Recipe.Spec"}, resourceType)
+}
+
+func getHookAndOpFromRecipe(recipe *Recipe.Recipe, name string) (*Recipe.Hook, *Recipe.Operation, error) {
+	// hook can be made up of optionalPrefix/hookName; workflow sequence uses full name
+	var prefix string
+
+	var suffix string
+
+	const containsSingleDelim = 2
+
+	parts := strings.Split(name, "/")
+	switch len(parts) {
+	case 1:
+		prefix = ""
+		suffix = name
+	case containsSingleDelim:
+		prefix = parts[0]
+		suffix = parts[1]
+	default:
+		return nil, nil, k8serrors.NewNotFound(schema.GroupResource{Resource: "Recipe.Spec.Hook.Name"}, name)
+	}
+
+	// match prefix THEN suffix
+	for _, hook := range recipe.Spec.Hooks {
+		if hook.Name == prefix {
+			for _, op := range hook.Ops {
+				if op.Name == suffix {
+					return hook, op, nil
+				}
+			}
+		}
+	}
+
+	return nil, nil, k8serrors.NewNotFound(schema.GroupResource{Resource: "Recipe.Spec.Hook.Name"}, name)
+}
+
+// TODO: complete functionality - add Hook support to KubeResourcesSpec, then copy in Velero object creation
+func convertRecipeHookToCaptureSpec(hook Recipe.Hook, op Recipe.Operation) (*ramen.KubeObjectsCaptureSpec, error) {
+	hookName := hook.Name + "-" + op.Name
```

Comment:
I wonder if there could be an out-of-space issue with really long names.  Maybe name can be replaced with a number to get a unique name like with the sub-backups?